s4e-backend/src/main/java/pl/cyfronet/s4e/data/repository/query/QueryDecorator.java

```diff
@@ -16,6 +16,11 @@ protected abstract void doPrepareQueryAndParameters(Map<String, Object> params,
                                                         StringBuilder resultQuery,
                                                         Errors errors);
 
+    protected abstract void doPrepareCountQueryAndParameters(Map<String, Object> params,
+                                                        List<Object> parameters,
+                                                        StringBuilder resultQuery,
+                                                        Errors errors);
```

Comment:
In all decorators, apart from QueryEnding, you just do the same as operations in this method as in doPrepareQueryAndParameters. I'd give this method a default implementation which calls doPrepareQueryAndParameters and only override it in QueryEnding. This would result in less repetition.